ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -126,24 +137,61 @@ private[state] object Conversions {
       DamlSubmissionDedupKey.SubmissionKind.CONFIGURATION,
     )
 
-  def buildSubmitterInfo(subInfo: SubmitterInfo): DamlSubmitterInfo =
-    DamlSubmitterInfo.newBuilder
+  def buildSubmitterInfo(subInfo: SubmitterInfo): DamlSubmitterInfo = {
+    val submitterInfoBuilder = DamlSubmitterInfo.newBuilder
       .addAllSubmitters((subInfo.actAs: List[String]).asJava)
       .setApplicationId(subInfo.applicationId)
       .setCommandId(subInfo.commandId)
-      .setDeduplicateUntil(
-        buildTimestamp(Time.Timestamp.assertFromInstant(subInfo.deduplicateUntil))
-      )
-      .build
+      .setSubmissionId(subInfo.submissionId)
+    subInfo.deduplicationPeriod match {
+      case DeduplicationPeriod.DeduplicationDuration(duration) =>
+        submitterInfoBuilder.setDeduplicationDuration(buildDuration(duration))
+      case DeduplicationPeriod.DeduplicationOffset(offset) =>
+        submitterInfoBuilder.setDeduplicationOffset(offset.toHexString)
+    }
+    submitterInfoBuilder.build
+  }
 
-  def parseSubmitterInfo(subInfo: DamlSubmitterInfo): SubmitterInfo =
-    SubmitterInfo(
+  def parseCompletionInfo(
+      recordTime: Instant,
+      subInfo: DamlSubmitterInfo,
+  ): CompletionInfo = {
+    val deduplicationPeriod = subInfo.getDeduplicationPeriodCase match {
+      case DeduplicationPeriodCase.DEDUPLICATION_DURATION =>
+        Some(
+          DeduplicationPeriod.DeduplicationDuration(parseDuration(subInfo.getDeduplicationDuration))
+        )
+      case DeduplicationPeriodCase.DEDUPLICATION_OFFSET =>
+        Some(
+          DeduplicationPeriod.DeduplicationOffset(
+            Offset.fromHexString(Ref.HexString.assertFromString(subInfo.getDeduplicationOffset))
+          )
+        )
+      case DeduplicationPeriodCase.DEDUPLICATE_UNTIL =>
+        //backwards compatibility
+        // we no longer support deduplicate until, and in v1 it was always relative to record time
+        // to keep backwards compatibility and simulate the new deduplication period we calculate the duration from the record time
```

Comment:
Feel free to ignore/edit, just a suggestion for improving the wording here:
```suggestion
        // For backwards compatibility with rejections generated by participant.state.v1 API.
        // As the deduplicate until timestamp is always relative to record time, we take the duration
        // between record time and the previous timestamp as the deduplication period (duration).
```